packages/legend-query-builder/src/components/watermark/QueryBuilderWatermark.tsx

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Dialog,
+  PanelDropZone,
+  PanelFormBooleanField,
+  PanelForm,
+  PanelDivider,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  PanelFormListItems,
+} from '@finos/legend-art';
+import {
+  isSuperType,
+  PrimitiveInstanceValue,
+  PRIMITIVE_TYPE,
+  type ValueSpecification,
+  type VariableExpression,
+} from '@finos/legend-graph';
+import { guaranteeNonNullable } from '@finos/legend-shared';
+import { observer } from 'mobx-react-lite';
+import { useCallback } from 'react';
+import { useDrop } from 'react-dnd';
+
+import {
+  type QueryBuilderParameterDragSource,
+  QUERY_BUILDER_PARAMETER_DND_TYPE,
+} from '../../stores/QueryBuilderParametersState.js';
+import type { QueryBuilderState } from '../../stores/QueryBuilderState.js';
+import type { QueryBuilderWatermarkState } from '../../stores/watermark/QueryBuilderWatermarkState.js';
+import { VariableExpressionViewer } from '../QueryBuilderParametersPanel.js';
+import { BasicValueSpecificationEditor } from '../shared/BasicValueSpecificationEditor.js';
+
+const WatermarkValueEditor = observer(
```

Comment:
can we isolate this component to assume the watermark value is defined. The modal component will handle the interaction of whether its defined or not. I would just make this component take as inputs
`watermarkValue: ValueSpec.., queryBuilderState: QueryBuilderState`

Reply:
I believe it needs the watermarkState still to call `setValue` on handleDrop but I can def remove the logic of defined or not and use the `watermarkValue` as input 👍